File: backend/src/main/java/com/niren/drama/entity/Character.java
package com.niren.drama.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("drama_character")
public class Character extends BaseEntity {
    private Long projectId;
    private String name;
    private String description;
    private String personality;
    private String appearance;
    /** male | female | other */
    private String gender;
    private String age;
    private String imageUrl;
    private String imageUrls;
    /** TTS voice identifier */
    private String voiceId;
    private String voiceName;
    /** TTS 语速，100=1.0x，可空用默认 1.0 */
    private Integer speechRate;
    /** 导演补充，合并进 TTS instruction */
    private String ttsNote;
    private Integer sortOrder;
}

File: frontend/src/api/character.ts
import request from './request'

export interface CharacterCreateParams {
  projectId: number | string
  name: string
  description?: string
  personality?: string
  appearance?: string
  gender?: string
  age?: string
  voiceId?: string
  voiceName?: string
}

export const characterApi = {
  create: (data: CharacterCreateParams) =>
    request.post('/characters', data),

  listByProject: (projectId: number | string) =>
    request.get(`/characters/project/${projectId}`),

  get: (id: number | string) =>
    request.get(`/characters/${id}`),

  update: (id: number | string, data: Partial<CharacterCreateParams>) =>
    request.put(`/characters/${id}`, data),

  delete: (id: number | string) =>
    request.delete(`/characters/${id}`),

  generateImage: (id: number | string) =>
    request.post(`/characters/${id}/generate-image`),
}
